HaruhiHeiretsuLib/Graphics/TextureFile.cs

```diff
@@ -230,6 +230,35 @@ public void Set20AF30Image(SKBitmap bitmap)
             Edited = true;
             switch (Mode)
             {
+                case ImageMode.IA8:
+                    int ia8Index = DataPointer;
+                    for (int y = 0; y < Height; y += 4)
+                    {
+                        int widthMod = (4 - (Width % 4)) == 4 ? 0 : 4 - (Width % 4);
+                        for (int x = 0; x < Width + widthMod; x += 4)
+                        {
+                            for (int row = 0; row < 4; row++)
+                            {
+                                for (int col = 0; col < 4; col++)
+                                {
+                                    if (ia8Index + 1 >= Data.Count || x + col >= Width || y + row >= Height)
+                                    {
+                                        ia8Index += 2;
+                                        continue;
+                                    }
+
+                                    SKColor color = bitmap.GetPixel(x + col, y + row);
+                                    byte grayscale = (byte)((color.Red / 3) + (color.Blue / 3) + (color.Green / 3));
```

Comment:
one last suggestion before you merge: personally i'd merge these lines with the below, so it becomes:

```csharp
Data[ia8Index] = color.Alpha;
Data[ia8Index + 1] = (byte)((color.Red / 3) + (color.Blue / 3) + (color.Green / 3));
```

but that's optional for sure; just makes the code a bit more condensed which I like